challenge_set_a_table/src/challenge_set_a_table.py

```diff
@@ -7,26 +7,64 @@
 import datetime
 import math
 
-from robot_smach_states import Initialize, Say, WaitTime
+from robot_smach_states import Initialize, Say
 from robot_smach_states.util.startup import startup
 import robot_smach_states.util.designators as ds
 
-#from challenge_set_a_table_states import detect
+# Still empty
+#from challenge_set_a_table_states import clean_table
+#from challenge_set_a_table_states import fetch_command
+#from challenge_set_a_table_states import set_table
 
 
 class ChallengeSetATable(smach.StateMachine):
     def __init__(self, robot):
         smach.StateMachine.__init__(self, outcomes=['Done','Aborted'])
 
         with self:
+            #Part I: Set a table
             smach.StateMachine.add('INITIALIZE',
                                    Initialize(robot),
+                                   transitions={'initialized': 'FETCH_COMMAND_I',
+                                                'abort': 'Aborted'})            
+            smach.StateMachine.add('FETCH_COMMAND_I', # Hear "set the table"
+                                   Initialize(robot),
+                                   transitions={'initialized': 'ASK_FOR_MEAL',
+                                                'abort': 'Aborted'})
+            smach.StateMachine.add('ASK_FOR_MEAL',
+                                   Say(robot, "What should I serve, master?"),
+                                   transitions={'spoken': 'SET_THE_TABLE'})
+
+            smach.StateMachine.add('SET_THE_TABLE', # Set the table (bring the objects to the table)
+                                   Initialize(robot),
+                                   transitions={'initialized': 'SERVE_MEAL',
+                                                'abort': 'Aborted'})
+            smach.StateMachine.add('SERVE_MEAL', # Serve the meal (pour milk into the bowl)
+                                   Initialize(robot),
+                                   transitions={'initialized': 'CORRECT_OBJECT_POSITIONS',
+                                                'abort': 'Aborted'})
+            smach.StateMachine.add('CORRECT_OBJECT_POSITIONS', # Inspect table and correct the moved objects
+                                   Initialize(robot),
+                                   transitions={'initialized': 'FETCH_COMMAND_II',
+                                                'abort': 'Aborted'})
+
+            #Part II: Clean the table
```

Comment:
Here is the full challenge. 
I have prepared empty files for part I and part II, which will be testable separately.